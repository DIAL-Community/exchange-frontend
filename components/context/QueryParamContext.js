import { createContext, useState } from 'react'

const QueryParamContext = createContext()
const QueryParamContextProvider = ({ children }) => {
  const [interactionDetected, setInteractionDetected] = useState(false)
  return (
    <QueryParamContext.Provider value={{ ...{ interactionDetected, setInteractionDetected } }}>
      {children}
    </QueryParamContext.Provider>
  )
}

export { QueryParamContextProvider, QueryParamContext }
