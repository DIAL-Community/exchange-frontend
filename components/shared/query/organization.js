import { gql } from '@apollo/client'

export const ORGANIZATION_SEARCH_QUERY = gql`
  query Organizations($search: String, $aggregatorOnly: Boolean) {
    organizations(search: $search, aggregatorOnly: $aggregatorOnly) {
      id
      name
      slug
      website
    }
  }
`

export const ORGANIZATION_CONTACT_QUERY = gql`
  query OrganizationOwners($slug: String!, $type: String!, $captcha: String!) {
    owners(slug: $slug, type: $type, captcha: $captcha) {
      email
    }
  }
`

export const ORGANIZATION_PAGINATION_ATTRIBUTES_QUERY = gql`
  query PaginationAttributeOrganization(
    $aggregatorOnly: Boolean
    $endorserOnly: Boolean
    $sectors: [String!]
    $countries: [String!]
    $years: [Int!]
    $search: String
  ) {
    paginationAttributeOrganization(
      aggregatorOnly: $aggregatorOnly
      endorserOnly: $endorserOnly
      sectors: $sectors
      countries: $countries
      years: $years
      search: $search
    ) {
      totalCount
    }
  }
`

export const PAGINATED_ORGANIZATIONS_QUERY = gql`
  query PaginatedOrganizations(
    $aggregatorOnly: Boolean
    $endorserOnly: Boolean
    $sectors: [String!]
    $countries: [String!]
    $years: [Int!]
    $search: String
    $limit: Int!
    $offset: Int!
  ) {
    paginatedOrganizations(
      aggregatorOnly: $aggregatorOnly
      endorserOnly: $endorserOnly
      sectors: $sectors
      countries: $countries
      years: $years
      search: $search
      offsetAttributes: { limit: $limit, offset: $offset }
    ) {
      id
      name
      slug
      imageFile
      parsedDescription
      organizationDescription {
        id
        description
        locale
      }
      sectors {
        id
      }
      countries {
        id
      }
      projects {
        id
      }
    }
  }
`

export const STOREFRONT_PAGINATION_ATTRIBUTES_QUERY = gql`
  query PaginationAttributeStorefront(
    $sectors: [String!]
    $countries: [String!]
    $buildingBlocks: [String!]
    $specialties: [String!]
    $certifications: [String!]
    $search: String!
  ) {
    paginationAttributeStorefront(
      sectors: $sectors
      countries: $countries
      buildingBlocks: $buildingBlocks
      specialties: $specialties
      certifications: $certifications
      search: $search
    ) {
      totalCount
    }
  }
`

export const PAGINATED_STOREFRONTS_QUERY = gql`
  query PaginatedStorefronts(
    $sectors: [String!]
    $countries: [String!]
    $buildingBlocks: [String!]
    $specialties: [String!]
    $certifications: [String!]
    $search: String!
    $limit: Int!
    $offset: Int!
  ) {
    paginatedStorefronts(
      sectors: $sectors
      countries: $countries
      buildingBlocks: $buildingBlocks
      specialties: $specialties
      certifications: $certifications
      search: $search
      offsetAttributes: { limit: $limit, offset: $offset }
    ) {
      id
      name
      slug
      imageFile
      parsedDescription
      organizationDescription {
        id
        description
        locale
      }
      sectors {
        id
      }
      countries {
        id
      }
      projects {
        id
      }
    }
  }
`

export const ORGANIZATION_DETAIL_QUERY = gql`
  query Organization($slug: String!) {
    organization(slug: $slug) {
      id
      name
      slug
      isMni
      website
      aliases
      imageFile
      isEndorser
      whenEndorsed
      endorserLevel
      hasStorefront
      haveOwner
      organizationDescription {
        id
        description
        locale
      }
      offices {
        id
        name
        slug
        province {
          id
          name
        }
        country {
          id
          name
          code
          codeLonger
        }
        cityData {
          id
          name
          slug
        }
        latitude
        longitude
      }
      sectors {
        id
        name
        slug
      }
      countries {
        id
        name
        slug
        code
      }
      products {
        id
        slug
        name
        imageFile
      }
      projects {
        id
        name
        slug
        organizations {
          id
          name
          slug
          imageFile
        }
        products {
          id
          name
          slug
          imageFile
        }
        origin {
          id
          slug
          name
        }
      }
      contacts {
        id
        name
        slug
        email
        title
      }
    }
  }
`

export const STOREFRONT_DETAIL_QUERY = gql`
  query Organization($slug: String!) {
    organization(slug: $slug) {
      id
      name
      slug
      website
      imageFile
      specialties
      hasStorefront
      heroFile
      organizationDescription {
        id
        description
        locale
      }
      offices {
        id
        name
        slug
        province {
          id
          name
        }
        country {
          id
          name
          code
          codeLonger
        }
        cityData {
          id
          name
          slug
        }
        latitude
        longitude
      }
      buildingBlockCertifications {
        id
        name
        slug
        imageFile
        category
        maturity
      }
      productCertifications {
        id
        name
        slug
        imageFile
      }
      sectors {
        id
        name
        slug
      }
      countries {
        id
        name
        slug
        code
      }
      projects {
        id
        name
        slug
        organizations {
          id
          name
          slug
          imageFile
        }
        products {
          id
          name
          slug
          imageFile
        }
        origin {
          slug
          name
        }
      }
      contacts {
        id
        name
        email
        title
      }
      resources {
        id
        name
        slug
        resourceLink
      }
    }
  }
`
