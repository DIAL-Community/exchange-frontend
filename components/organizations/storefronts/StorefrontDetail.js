import { useQuery } from '@apollo/client'
import { useEffect, useRef } from 'react'
import NotFound from '../../shared/NotFound'
import { Error, Loading } from '../../shared/FetchStatus'
import { ORGANIZATION_QUERY } from '../../../queries/organization'
import StorefrontDetailLeft from './StorefrontDetailLeft'
import StorefrontDetailRight from './StorefrontDetailRight'

const StorefrontDetail = ({ slug, locale }) => {
  const commentsSectionElement = useRef()

  const { loading, error, data, refetch } = useQuery(ORGANIZATION_QUERY, {
    variables: { slug },
    context: { headers: { 'Accept-Language': locale } },
    skip: !slug
  })

  useEffect(() => {
    refetch()
  }, [locale, refetch])

  if (loading) {
    return <Loading />
  } else if (error) {
    return <Error />
  } else if (!data?.organization || !data.organization.hasStorefront) {
    return <NotFound />
  }

  return (
    <>
      {
        data?.organization &&
          <div className='flex flex-col lg:flex-row justify-between pb-8'>
            <div className='relative lg:sticky lg:top-66px w-full lg:w-1/3 xl:w-1/4 h-full py-4 px-4'>
              <StorefrontDetailLeft
                organization={data.organization}
                commentsSectionRef={commentsSectionElement}
              />
            </div>
            <div className='w-full lg:w-2/3 xl:w-3/4'>
              <StorefrontDetailRight
                organization={data.organization}
                commentsSectionRef={commentsSectionElement}
              />
            </div>
          </div>
      }
    </>
  )
}

export default StorefrontDetail
