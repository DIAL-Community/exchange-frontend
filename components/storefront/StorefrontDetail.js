import { useRef } from 'react'
import { useQuery } from '@apollo/client'
import { GRAPH_QUERY_CONTEXT } from '../../lib/apolloClient'
import Breadcrumb from '../shared/Breadcrumb'
import { handleLoadingQuery, handleMissingData, handleQueryError } from '../shared/GraphQueryHandler'
import { STOREFRONT_DETAIL_QUERY } from '../shared/query/organization'
import StorefrontDetailLeft from './StorefrontDetailLeft'
import StorefrontDetailRight from './StorefrontDetailRight'

const StorefrontDetail = ({ slug }) => {
  const scrollRef = useRef(null)

  const { loading, error, data } = useQuery(STOREFRONT_DETAIL_QUERY, {
    variables: { slug },
    context: {
      headers: {
        ...GRAPH_QUERY_CONTEXT.VIEWING
      }
    }
  })

  if (loading) {
    return handleLoadingQuery()
  } else if (error) {
    return handleQueryError(error)
  } else if (!data?.organization) {
    return handleMissingData()
  }

  const { organization } = data

  const slugNameMapping = (() => {
    const map = {}
    map[organization.slug] = organization.name

    return map
  })()

  return (
    <div className='lg:px-8 xl:px-56 flex flex-col'>
      <div className='px-4 lg:px-6 py-4 bg-dial-violet text-dial-stratos ribbon-detail z-40'>
        <Breadcrumb slugNameMapping={slugNameMapping}/>
      </div>
      <div className='flex flex-col lg:flex-row gap-x-8'>
        <div className='lg:basis-1/3'>
          <StorefrontDetailLeft scrollRef={scrollRef} organization={organization} />
        </div>
        <div className='lg:basis-2/3'>
          <StorefrontDetailRight ref={scrollRef} organization={organization} />
        </div>
      </div>
    </div>
  )
}

export default StorefrontDetail
