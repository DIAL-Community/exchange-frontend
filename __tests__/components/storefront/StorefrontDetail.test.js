import { screen } from '@testing-library/dom'
import userEvent from '@testing-library/user-event'
import { QueryParamContextProvider } from '../../../components/context/QueryParamContext'
import { CREATE_ORGANIZATION } from '../../../components/shared/mutation/organization'
import { COMMENTS_QUERY } from '../../../components/shared/query/comment'
import {
  ORGANIZATION_DETAIL_QUERY, PAGINATED_STOREFRONTS_QUERY, STOREFRONT_PAGINATION_ATTRIBUTES_QUERY
} from '../../../components/shared/query/organization'
import StorefrontEdit from '../../../components/storefront/StorefrontEdit'
import { render } from '../../test-utils'
import CustomMockedProvider, { generateMockApolloData } from '../../utils/CustomMockedProvider'
import { mockNextAuthUseSession, mockNextUseRouter, mockTenantApi } from '../../utils/nextMockImplementation'
import { commentsQuery, createStorefront, storefrontDetail } from './data/StorefrontDetail.data'
import { paginatedStorefronts, storefrontPaginationAttribute } from './data/StorefrontMain.data'

mockTenantApi()
mockNextUseRouter()
describe('Unit tests for the storefront detail page.', () => {
  const mockStorefront = generateMockApolloData(
    ORGANIZATION_DETAIL_QUERY,
    {
      'slug': 'ai4gov'
    },
    null,
    storefrontDetail
  )

  const mockStorefrontComments = generateMockApolloData(
    COMMENTS_QUERY,
    {
      'commentObjectId': 190,
      'commentObjectType': 'STOREFRONT'
    },
    null,
    commentsQuery
  )

  test('Should render detail of a storefront.', async () => {
    const { container } = render(
      <CustomMockedProvider mocks={[mockStorefront, mockStorefrontComments]}>
        <QueryParamContextProvider>
          <StorefrontEdit slug='ai4gov' />
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )

    expect(await screen.findByText('AI4GOV')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  test('Should render unauthorized for non logged in user.', async () => {
    const { container } = render(
      <CustomMockedProvider mocks={[mockStorefront, mockStorefrontComments]}>
        <QueryParamContextProvider>
          <StorefrontEdit slug='ai4gov' />
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )

    expect(await screen.findByText('AI4GOV')).toBeInTheDocument()
    expect(await screen.findByText('You are not authorized to view this page')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  test('Should render edit page for logged in user.', async () => {
    mockNextAuthUseSession()

    const mockCreateBuildingBlock = generateMockApolloData(
      CREATE_ORGANIZATION,
      {
        'name': 'AI4GOV - Edited',
        'slug': 'ai4gov',
        'aliases': [''],
        'website': 'www.ai4gov.net',
        'description': 'Description for the organization.',
        'hasStorefront': true
      },
      null,
      createStorefront
    )

    const mockStorefrontPaginationAttribute = generateMockApolloData(
      STOREFRONT_PAGINATION_ATTRIBUTES_QUERY,
      { search: '' },
      null,
      storefrontPaginationAttribute
    )

    const mockPaginatedStorefronts = generateMockApolloData(
      PAGINATED_STOREFRONTS_QUERY,
      { search: '', limit: 8, offset: 0 },
      null,
      paginatedStorefronts
    )

    const { container } = render(
      <CustomMockedProvider
        mocks={[
          mockStorefront,
          mockStorefrontComments,
          mockCreateBuildingBlock,
          mockStorefrontPaginationAttribute,
          mockPaginatedStorefronts,
          mockStorefront
        ]}
      >
        <QueryParamContextProvider>
          <StorefrontEdit slug='ai4gov' />
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )

    expect(await screen.findByText('AI4GOV')).toBeInTheDocument()

    const repositoryNameInput = screen.getByDisplayValue('AI4GOV')
    expect(repositoryNameInput.value).toBe('AI4GOV')

    const user = userEvent.setup()
    await user.type(repositoryNameInput, ' - Edited')
    expect(repositoryNameInput.value).toBe('AI4GOV - Edited')

    const repositorySubmitButton = screen.getByText('Submit Storefront')
    await user.click(repositorySubmitButton)

    expect(container).toMatchSnapshot()
  })
})
