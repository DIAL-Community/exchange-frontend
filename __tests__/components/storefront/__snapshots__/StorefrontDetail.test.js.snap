// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for the storefront detail page. Should render detail of a storefront. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="lg:px-8 xl:px-56 flex flex-col"
  >
    <div
      class="px-4 lg:px-6 py-4 bg-dial-blue-chalk text-dial-stratos ribbon-detail z-40"
    >
      <div
        class="whitespace-nowrap text-ellipsis overflow-hidden"
      >
        <a
          class="h5"
          href="/"
        >
          Home
        </a>
      </div>
    </div>
    <div
      class="flex flex-col lg:flex-row gap-x-8"
    >
      <div
        class="lg:basis-1/3"
      >
        <div
          class="bg-dial-slate-100 lg:h-full"
        >
          <div
            class="flex flex-col gap-y-3 px-4 lg:px-6 lg:py-3"
          >
            <div
              class="flex flex-col gap-y-4 py-3"
            >
              <div
                class="text-xl text-dial-plum font-semibold"
              >
                AI4GOV
              </div>
              <div
                class="flex justify-center items-center py-16 bg-white rounded border"
              >
                <div
                  class="inline"
                >
                  <img
                    alt="Logo of Storefront."
                    class="object-contain w-20 h-20"
                    src="http://localhost:3000//assets/organizations/ai4gov.png"
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-y-8 text-sm pt-6 pb-3"
              >
                <div
                  class="flex flex-col gap-y-3"
                >
                  <div
                    class="font-semibold text-dial-sapphire"
                  >
                    Website
                  </div>
                  <div
                    class="flex text-dial-stratos"
                  >
                    <a
                      class="flex border-b border-dial-iris-blue "
                      href="//www.ai4gov.net"
                      rel="noreferrer"
                      target="_blank"
                    >
                      <div
                        class="line-clamp-1 break-all"
                      >
                        www.ai4gov.net
                      </div>
                    </a>
                     ⧉
                  </div>
                </div>
                <div
                  class="flex flex-col gap-y-3"
                >
                  <div
                    class="flex flex-row gap-3"
                  >
                    <div
                      class="font-semibold text-dial-sapphire"
                    >
                      Sectors
                    </div>
                    <div
                      class="flex gap-3 ml-auto"
                    />
                  </div>
                  <div
                    class="flex flex-col gap-y-2"
                  >
                    <div>
                      <div>
                        Digital, Data, and Technology
                      </div>
                    </div>
                    <div>
                      <div>
                        Education and Social Development
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
          </div>
        </div>
      </div>
      <div
        class="lg:basis-2/3"
      >
        <form>
          <div
            class="px-4 lg:px-0 py-4 lg:py-6 text-dial-plum"
          >
            <div
              class="flex flex-col gap-y-6 text-sm"
            >
              <div
                class="text-xl font-semibold"
              >
                Edit AI4GOV
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire required-field"
                  for="name"
                >
                  Name
                </label>
                <input
                  class="w-full text-sm"
                  id="name"
                  name="name"
                  placeholder="Name"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire"
                >
                  Aliases
                </label>
                <div
                  class="flex gap-x-2"
                >
                  <input
                    class="w-full text-sm"
                    name="aliases.0.value"
                    placeholder="Alias"
                    type="text"
                    value=""
                  />
                  <span>
                    <button
                      class="shadow-md px-3 py-2.5 hover:bg-dial-angel"
                      type="button"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 448 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
                        />
                      </svg>
                    </button>
                  </span>
                </div>
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire required-field"
                  for="website"
                >
                  Website
                </label>
                <div
                  class="border border-transparent url-input bg-white flex items-center"
                >
                  <span
                    class="select-none text-dial-gray"
                  >
                    https://
                  </span>
                  <input
                    class="url w-full text-sm placeholder-dial-gray border-0 py-0 px-0 focus:outline-0"
                    id="website"
                    placeholder="Website"
                    value="www.ai4gov.net"
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire"
                >
                  Logo
                </label>
                <label
                  class="flex items-center"
                >
                  <input
                    accept=".jpg,.jpeg,.png,.bmp"
                    class="bg-white file:hidden w-full text-sm"
                    name="imageFile"
                    type="file"
                  />
                  <div
                    class="-ml-11 flex p-1"
                  >
                    <span
                      class="h-full pl-1 border-l text-dial-gray-dark border-dial-gray-dark"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        version="1.1"
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M13 14h-10v-2l3-5 4.109 5 2.891-2v4z"
                        />
                        <path
                          d="M13 7.5c0 0.828-0.672 1.5-1.5 1.5s-1.5-0.672-1.5-1.5 0.672-1.5 1.5-1.5c0.828 0 1.5 0.672 1.5 1.5z"
                        />
                        <path
                          d="M14.341 3.579c-0.347-0.473-0.831-1.027-1.362-1.558s-1.085-1.015-1.558-1.362c-0.806-0.591-1.197-0.659-1.421-0.659h-7.75c-0.689 0-1.25 0.561-1.25 1.25v13.5c0 0.689 0.561 1.25 1.25 1.25h11.5c0.689 0 1.25-0.561 1.25-1.25v-9.75c0-0.224-0.068-0.615-0.659-1.421zM12.271 2.729c0.48 0.48 0.856 0.912 1.134 1.271h-2.406v-2.405c0.359 0.278 0.792 0.654 1.271 1.134zM14 14.75c0 0.136-0.114 0.25-0.25 0.25h-11.5c-0.135 0-0.25-0.114-0.25-0.25v-13.5c0-0.135 0.115-0.25 0.25-0.25 0 0 7.749-0 7.75 0v3.5c0 0.276 0.224 0.5 0.5 0.5h3.5v9.75z"
                        />
                      </svg>
                    </span>
                  </div>
                </label>
                <div
                  class="text-xs italic text-dial-stratos"
                >
                  Note: Supported file types are jpg, png, and bmp
                </div>
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire required-field"
                >
                  Description
                </label>
                <div
                  class="relative"
                >
                  <svg
                    class="absolute text-lg inset-x-1/2 top-10 spinner"
                    fill="currentColor"
                    height="2em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="2em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M304 48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zm0 416a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM48 304a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm464-48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM142.9 437A48 48 0 1 0 75 369.1 48 48 0 1 0 142.9 437zm0-294.2A48 48 0 1 0 75 75a48 48 0 1 0 67.9 67.9zM369.1 437A48 48 0 1 0 437 369.1 48 48 0 1 0 369.1 437z"
                    />
                  </svg>
                  <div
                    class="html-editor"
                  >
                    <textarea
                      id="description-editor"
                      style="visibility: hidden;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="flex flex-wrap text-base mt-6 gap-3"
              >
                <button
                  class="submit-button"
                  type="submit"
                >
                  Submit Storefront
                </button>
                <button
                  class="cancel-button"
                  type="button"
                >
                  Cancel
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the storefront detail page. Should render edit page for logged in user. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="lg:px-8 xl:px-56 flex flex-col"
  >
    <div
      class="px-4 lg:px-6 py-4 bg-dial-blue-chalk text-dial-stratos ribbon-detail z-40"
    >
      <div
        class="whitespace-nowrap text-ellipsis overflow-hidden"
      >
        <a
          class="h5"
          href="/"
        >
          Home
        </a>
      </div>
    </div>
    <div
      class="flex flex-col lg:flex-row gap-x-8"
    >
      <div
        class="lg:basis-1/3"
      >
        <div
          class="bg-dial-slate-100 lg:h-full"
        >
          <div
            class="flex flex-col gap-y-3 px-4 lg:px-6 lg:py-3"
          >
            <div
              class="flex flex-col gap-y-4 py-3"
            >
              <div
                class="text-xl text-dial-plum font-semibold"
              >
                AI4GOV - Edited
              </div>
              <div
                class="flex justify-center items-center py-16 bg-white rounded border"
              >
                <div
                  class="inline"
                >
                  <img
                    alt="Logo of Storefront."
                    class="object-contain w-20 h-20"
                    src="http://localhost:3000//assets/organizations/ai4gov.png"
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-y-8 text-sm pt-6 pb-3"
              >
                <div
                  class="flex flex-col gap-y-3"
                >
                  <div
                    class="font-semibold text-dial-sapphire"
                  >
                    Website
                  </div>
                  <div
                    class="flex text-dial-stratos"
                  >
                    <a
                      class="flex border-b border-dial-iris-blue "
                      href="//www.ai4gov.net"
                      rel="noreferrer"
                      target="_blank"
                    >
                      <div
                        class="line-clamp-1 break-all"
                      >
                        www.ai4gov.net
                      </div>
                    </a>
                     ⧉
                  </div>
                </div>
                <div
                  class="flex flex-col gap-y-3"
                >
                  <div
                    class="flex flex-row gap-3"
                  >
                    <div
                      class="font-semibold text-dial-sapphire"
                    >
                      Sectors
                    </div>
                    <div
                      class="flex gap-3 ml-auto"
                    >
                      <button
                        class="cursor-pointer bg-dial-iris-blue px-2 py-1 rounded text-white"
                        type="button"
                      >
                        <svg
                          class="inline pb-0.5"
                          fill="none"
                          height="1em"
                          stroke="currentColor"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-width="2"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M12 20h9"
                          />
                          <path
                            d="M16.5 3.5a2.121 2.121 0 0 1 3 3L7 19l-4 1 1-4L16.5 3.5z"
                          />
                        </svg>
                        <span
                          class="text-sm px-1"
                        >
                          Edit
                        </span>
                      </button>
                    </div>
                  </div>
                  <div
                    class="flex flex-col gap-y-2"
                  >
                    <div>
                      <div>
                        Digital, Data, and Technology
                      </div>
                    </div>
                    <div>
                      <div>
                        Education and Social Development
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
          </div>
        </div>
      </div>
      <div
        class="lg:basis-2/3"
      >
        <form>
          <div
            class="px-4 lg:px-0 py-4 lg:py-6 text-dial-plum"
          >
            <div
              class="flex flex-col gap-y-6 text-sm"
            >
              <div
                class="text-xl font-semibold"
              >
                Edit AI4GOV - Edited
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire required-field"
                  for="name"
                >
                  Name
                </label>
                <input
                  class="w-full text-sm"
                  id="name"
                  name="name"
                  placeholder="Name"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire"
                >
                  Aliases
                </label>
                <div
                  class="flex gap-x-2"
                >
                  <input
                    class="w-full text-sm"
                    name="aliases.0.value"
                    placeholder="Alias"
                    type="text"
                    value=""
                  />
                  <span>
                    <button
                      class="shadow-md px-3 py-2.5 hover:bg-dial-angel"
                      type="button"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 448 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
                        />
                      </svg>
                    </button>
                  </span>
                </div>
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire required-field"
                  for="website"
                >
                  Website
                </label>
                <div
                  class="border border-transparent url-input bg-white flex items-center"
                >
                  <span
                    class="select-none text-dial-gray"
                  >
                    https://
                  </span>
                  <input
                    class="url w-full text-sm placeholder-dial-gray border-0 py-0 px-0 focus:outline-0"
                    id="website"
                    placeholder="Website"
                    value="www.ai4gov.net"
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire"
                >
                  Logo
                </label>
                <label
                  class="flex items-center"
                >
                  <input
                    accept=".jpg,.jpeg,.png,.bmp"
                    class="bg-white file:hidden w-full text-sm"
                    name="imageFile"
                    type="file"
                  />
                  <div
                    class="-ml-11 flex p-1"
                  >
                    <span
                      class="h-full pl-1 border-l text-dial-gray-dark border-dial-gray-dark"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        version="1.1"
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M13 14h-10v-2l3-5 4.109 5 2.891-2v4z"
                        />
                        <path
                          d="M13 7.5c0 0.828-0.672 1.5-1.5 1.5s-1.5-0.672-1.5-1.5 0.672-1.5 1.5-1.5c0.828 0 1.5 0.672 1.5 1.5z"
                        />
                        <path
                          d="M14.341 3.579c-0.347-0.473-0.831-1.027-1.362-1.558s-1.085-1.015-1.558-1.362c-0.806-0.591-1.197-0.659-1.421-0.659h-7.75c-0.689 0-1.25 0.561-1.25 1.25v13.5c0 0.689 0.561 1.25 1.25 1.25h11.5c0.689 0 1.25-0.561 1.25-1.25v-9.75c0-0.224-0.068-0.615-0.659-1.421zM12.271 2.729c0.48 0.48 0.856 0.912 1.134 1.271h-2.406v-2.405c0.359 0.278 0.792 0.654 1.271 1.134zM14 14.75c0 0.136-0.114 0.25-0.25 0.25h-11.5c-0.135 0-0.25-0.114-0.25-0.25v-13.5c0-0.135 0.115-0.25 0.25-0.25 0 0 7.749-0 7.75 0v3.5c0 0.276 0.224 0.5 0.5 0.5h3.5v9.75z"
                        />
                      </svg>
                    </span>
                  </div>
                </label>
                <div
                  class="text-xs italic text-dial-stratos"
                >
                  Note: Supported file types are jpg, png, and bmp
                </div>
              </div>
              <div
                class="flex flex-col gap-y-2"
              >
                <label
                  class="text-dial-sapphire required-field"
                >
                  Description
                </label>
                <div
                  class="relative"
                >
                  <svg
                    class="absolute text-lg inset-x-1/2 top-10 spinner"
                    fill="currentColor"
                    height="2em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="2em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M304 48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zm0 416a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM48 304a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm464-48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM142.9 437A48 48 0 1 0 75 369.1 48 48 0 1 0 142.9 437zm0-294.2A48 48 0 1 0 75 75a48 48 0 1 0 67.9 67.9zM369.1 437A48 48 0 1 0 437 369.1 48 48 0 1 0 369.1 437z"
                    />
                  </svg>
                  <div
                    class="html-editor"
                  >
                    <textarea
                      id="description-editor"
                      style="visibility: hidden;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="flex flex-wrap text-base mt-6 gap-3"
              >
                <button
                  class="submit-button"
                  disabled=""
                  type="submit"
                >
                  Submit Storefront
                  <svg
                    class="spinner ml-3"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M304 48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zm0 416a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM48 304a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm464-48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM142.9 437A48 48 0 1 0 75 369.1 48 48 0 1 0 142.9 437zm0-294.2A48 48 0 1 0 75 75a48 48 0 1 0 67.9 67.9zM369.1 437A48 48 0 1 0 437 369.1 48 48 0 1 0 369.1 437z"
                    />
                  </svg>
                </button>
                <button
                  class="cancel-button"
                  disabled=""
                  type="button"
                >
                  Cancel
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the storefront detail page. Should render unauthorized for non logged in user. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="min-h-[75vh] flex items-center justify-center"
  >
    <div
      class="flex flex-col gap-y-6"
    >
      <svg
        class="spinner mx-auto"
        fill="currentColor"
        height="3em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="3em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M304 48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zm0 416a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM48 304a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm464-48a48 48 0 1 0 -96 0 48 48 0 1 0 96 0zM142.9 437A48 48 0 1 0 75 369.1 48 48 0 1 0 142.9 437zm0-294.2A48 48 0 1 0 75 75a48 48 0 1 0 67.9 67.9zM369.1 437A48 48 0 1 0 437 369.1 48 48 0 1 0 369.1 437z"
        />
      </svg>
      <div
        class="text-center"
      >
        Processing request ...
      </div>
    </div>
  </div>
</div>
`;
